### ISSUE TITLE
Census Custom Estimator: Found more than one metagraph event per run

### ISSUE THREAD

**Issue by jcomfort4** [GoogleCloudPlatform/cloudml-samples#73]
For problems running the sample code please provide the following information.

### System information
- Mac OS
- 1.2.1 (to be compatible with ML Engine
- 2.7
- gcloud ml-engine local train --package-path trainer --module-name trainer.task                            --                            --train-files $TRAIN_FILE --eval-files $EVAL_FILE --job-dir $OUTPUT_DIR --train-steps $TRAIN_STEPS  --eval-steps 100

then tensorboard --logdir=$OUTPUT_DIR

### Describe the problem
When I run the sample code for the custom estimator and view the tensorboard, I am getting the warning "WARNING:tensorflow:Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events.  Overwriting the graph with the newest event.
WARNING:tensorflow:Found more than one metagraph event per run. Overwriting the metagraph with the newest event." and I am trying to understand how to get rid of this as it seems like it is indicative of some behavior that is not desirable. Any ideas why the code is generating multiple events per run? I imagine it is because tf.summary.scalar('loss', loss) is being written for both TRAIN/EVAL but then there are also eval_metric_ops?

**Comment by elibixby** [comment]
Are you running multiple iterations with the same output dir? If you are manually writing out other metrics in the same directory you will get this error. Estimator takes care of "scoping" it's eval and train metrics in directories (train is in the root, eval is in an `eval/` subdirectory). Need more details from you as this isn't reproducable for us.

**Comment by jcomfort4** [comment]
I am deleting the output dir before every run with `rm -rf $OUTPUT_DIR` and I am still getting this error.  I have double (and triple) checked that output has been deleted in its entirety before starting a new run and I still seem to get this warning.  I have even inspected afterwards and I see train in the root and eval/ in subdirectory as you describe which is why I'm confused as to what is going on.

Are you not able to reproduce even using the current custom estimator census code? Here is my exact sequence of steps that I just repeated and received the warning again.

```
customestimator$ ls
config.yaml	trainer
```
NOTE: the above shows there is no output directory
```
MacBook-Pro:customestimator $ gcloud ml-engine local train --package-path trainer \
> --module-name trainer.task \
> -- \
> --train-files gs://cloudml-public/census/data/adult.data.csv \
> --eval-files gs://cloudml-public/census/data/adult.test.csv \
> --job-dir output \
> --train-steps 1000 \
> --eval-steps 100

```

This trained succesfully for 1000 steps and then I performed
```
customestimator $ tensorboard --logdir=output
WARNING:tensorflow:Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events.  Overwriting the graph with the newest event.
WARNING:tensorflow:Found more than one metagraph event per run. Overwriting the metagraph with the newest event.
```

Is there any step that I am missing? When I go inspect the output directory I have the following contents as expected, with a second events file being located in eval/.

```
output comfort$ ls
checkpoint
eval
events.out.tfevents.1504639056.Jimmys-MacBook-Pro.local
export
graph.pbtxt
model.ckpt-1.data-00000-of-00001
model.ckpt-1.index
model.ckpt-1.meta
model.ckpt-1000.data-00000-of-00001
model.ckpt-1000.index
model.ckpt-1000.meta
```


At training time the only warning I get is the warning about Monitors being deprecated, but I don't believe there should be any issues.
```
WARNING:tensorflow:From /Users/comfort/anaconda/lib/python2.7/site-packages/tensorflow/contrib/learn/python/learn/monitors.py:268: __init__ (from tensorflow.contrib.learn.python.learn.monitors) is deprecated and will be removed after 2016-12-05.
Instructions for updating:
Monitors are deprecated. Please use tf.train.SessionRunHook.
```

**Comment by puneith** [comment]
@jcomfort4 Are you getting this WARNING in the newest TF1.4 sample and TensorBoard as well?

**Comment by puneith** [comment]
@jcomfort4 Haven't heard back on the status of new samples and TF version so closing this. Feel free to reopen or file another issue.

**Comment by jcomfort4** [comment]
@puneith Sorry that I missed this the first time. I'm using TF 1.6 and I just upgraded tensorboard via pip install to ensure I was using the newest tensorboard, but I am still getting this error:

W0502 06:41:40.734479 Reloader plugin_event_accumulator.py:300] Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events.  Overwriting the graph with the newest event.
W0502 06:41:40.748988 Reloader plugin_event_accumulator.py:308] Found more than one metagraph event per run. Overwriting the metagraph with the newest event.

The tensorboard appears fine (graph looks correct, results logged properly, etc.) and I don't see anything missing, but still not sure why I'm seeing this error.

**Comment by joellerena** [comment]
Regards.
I have the same message:
WARNING: tensorflow: path ../external/data/plugin/text/runs not found, sending 404
WARNING: tensorflow: Found more than one graph event per run, or there was a metagraph containing a graph_def, as well as one or more graph events. Overwriting the graph with the newest event.

ipython==5.8.0
matplotlib==2.2.2
numpy==1.15.0
pandas==0.23.3
protobuf==3.6.1
python-dateutil==2.7.3
scikit-learn==0.19.1
scipy==1.1.0
tensorboard==1.11.0
tensorflow-gpu==1.1.0

**Comment by Kublai-Jing** [comment]
Is this issue resolved ? Having the same issue when using custom estimator

**Comment by joellerena** [comment]
Solucionado, gracias.
